src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -150,6 +150,7 @@ private static IArgument DeriveArgument(int parameterIndex, int argumentIndex, I
                 (argument) =>
                 {
                     string name = !argumentNames.IsDefaultOrEmpty ? argumentNames[argumentIndex] : null;
+                    Symbols.ParameterSymbol parameter = parameterIndex < parameters.Length ? parameters[parameterIndex] : null;
```

Comment:
Relying on HasErrors worries me. The more I think about this issue the more worried I get. My current belief:

Implementations of IOperation should treat the bound trees as unreliable in the sense that they are not guaranteed to obey a structural contract. Properties that one wouldn't expect to ever be null (e.g. the operands of a binary operator) should be tested for nullness. Arrays shouldn't be trusted to be of the expected size. Etc.

Further, an IOperation implementation should obey a structural contract, because otherwise analyzers will crash. This implies introducing Invalid objects where the bound trees do not supply valid values. For example, if an operand of a binary operator is missing in the bound trees, an IOperation implementation should supply an Invalid operand. If a symbol reference in the bound trees does not bind to a symbol, an IOperation implementation should supply an Invalid symbol of some kind.

If we don't do this either IOperation implementations or analyzers that use them will be subject to random crashes when a compiler produces a malformed bound tree.

There isn't time to do this hardening for Update 2.